core/commands/tags.py

```diff
@@ -0,0 +1,318 @@
+import os
+
+import sublime
+from sublime_plugin import WindowCommand, TextCommand, EventListener
+
+from ..git_command import GitCommand
+from ...common import util
+
+TAG_DELETE_MESSAGE = "Tag deleted."
+TAG_CREATE_PROMPT = "Enter tag:"
+TAG_CREATE_MESSAGE_PROMPT = "Enter message:"
+START_PUSH_MESSAGE = "Starting push..."
+END_PUSH_MESSAGE = "Push complete."
+NO_REMOTES_MESSAGE = "You have not configured any remotes."
+
+VIEW_TITLE = "TAGS: {}"
+
+LOCAL_TEMPLATE = """
+  LOCAL:
+{}
+"""
+
+REMOTE_TEMPLATE = """
+  REMOTE ({}):
+{}
+"""
+
+VIEW_HEADER_TEMPLATE = """
+  BRANCH:  {branch_status}
+  ROOT:    {repo_root}
+  HEAD:    {current_head}
+"""
+
+NO_LOCAL_TAGS_MESSAGE = "    Your repository has no tags."
+NO_REMOTE_TAGS_MESSAGE = "    This remote has no tags."
+
+KEY_BINDINGS_MENU = """
+  #############
+  ## ACTIONS ##
+  #############
+
+  [c] create
+  [d] delete
+  [p] push to remote
+  [P] push all tags to remote
+  [l] view commit
+
+  ###########
+  ## OTHER ##
+  ###########
+
+  [r] refresh status
+
+-
+"""
+
+view_section_ranges = {}
+
+
+class GsShowTagsCommand(WindowCommand, GitCommand):
+
+    """
+    Open a tags view for the active git repository.
+    """
+
+    def run(self):
+        repo_path = self.repo_path
+        title = VIEW_TITLE.format(os.path.basename(repo_path))
+        tags_view = util.view.get_read_only_view(self, "tags")
+        util.view.disable_other_plugins(tags_view)
+        tags_view.set_name(title)
+        tags_view.set_syntax_file("Packages/GitSavvy/syntax/tags.tmLanguage")
+        tags_view.settings().set("git_savvy.repo_path", repo_path)
+        tags_view.settings().set("word_wrap", False)
+        self.window.focus_view(tags_view)
+        tags_view.sel().clear()
+
+        tags_view.run_command("gs_tags_refresh")
+
+
+class GsTagsRefreshCommand(TextCommand, GitCommand):
+
+    """
+    Get the current state of the git repo and display tags and command
+    menu to the user.
+    """
+
+    def run(self, edit, **kwargs):
+        sublime.set_timeout_async(lambda: self.run_async(**kwargs))
```

Comment:
That is pretty much copied from the status view.
https://github.com/divmain/GitSavvy/blob/7713415730f8daa5907704695415b00e66bb74c1/core/commands/status.py#L118-L119